internal/autoevent/executor.go

```diff
@@ -47,14 +47,21 @@ func (e *Executor) Run(ctx context.Context, wg *sync.WaitGroup, buffer chan bool
 	defer wg.Done()
 
 	lc := bootstrapContainer.LoggingClientFrom(dic.Get)
+	deadline := time.Now().Add(e.duration)
+
 	for {
 		select {
 		case <-ctx.Done():
 			return
-		case <-time.After(e.duration):
+		default:
```

Comment:
Don't use `default` because the go routine cannot break when ctx.Done() happens.  It has to wait for another loop.